### DISCUSSION TITLE
Preferred music source?

### DISCUSSION THREAD

**Discussion by mzurhorst** [music-assistant/support#2391] [Q&A] [answered]
Hi all,

In my setup, I have ~500 albums on a local NFS share.  This share is mounted into a local Plex server, which works great to stream music remote to my phone.  At the same time, the NFS is also mounted into Home Assistant.

Accidentally, I have now all albums twice in my Music Assistant library.
This makes we wonder which of the two sources should be the preference? 

Is there any criteria which helps me deciding? -- Performance? -- FanArt?

Thank you.

**Comment by OzGav** [comment]
My first thought is to remove the Plex provider. Why have both?

**Comment by mzurhorst** [comment]
I am not sure, honestly.   I was poking around in the settings for a few months and did not manage to get sound played.  I was testing every now and than after an upgrade on the home assistant plugin.

For quite a while, I was not able to browse the local folders.  Thus, I assume this is a left-over from my trial & error phase.


I now removed the local folders provider in the settings, but its items still appear in the library.
I was thinking that I could have an advantage when sticking to Plex with regard to recently-played history or other statistics captured in Plex.


How can I remove the duplicates? -- Is there a way to delete the library and re-scan from scratch?

**Reply by OzGav** [reply]
Removing the provider should remove all associated entries. There is an open issue due to some being left behind. There usually aren’t many so you can manually “remove from library” those. You might be running into the problem whereby items won’t be removed if they have been linked to the other provider although in this case you shouldn’t see duplicates.

You always have the nuclear option of removing the addon or the docker container and starting again but we want to fix it so you don’t have to do that but I am not sure when that issue iI mentioned is due for resolution.

**Reply by mzurhorst** [reply]
Maybe there is a missunderstanding about the icons.
But I have now done a lot of checks, and all the albums I have checks have the tracks listed twice.

I was under the impression that "This item is available in the library" would be the local folders.


When I sort alphabetically, it looks like this:  
![grafik](https://github.com/music-assistant/hass-music-assistant/assets/575190/79c0eea4-a93f-4a0a-a0db-ee9b7583d303)

**Reply by OzGav** [reply, marked as answer]
No that is a bug and subject to this issue https://github.com/music-assistant/hass-music-assistant/issues/2319

Items in library is not just local files. Yes local files are added to the library by default but so are a lot of other items. You need to review the docs for each provider. For example, all items in the Spotify "Your Library" will be added.
